MonoGame.Framework/Windows/WinFormsGameWindow.cs

```diff
@@ -508,6 +508,11 @@ void Dispose(bool disposing)
                     _form = null;
                 }
             }
+            _platform = null;
+            Game = null;
+            Mouse.SetWindows(null);
+            Device.KeyboardInput -= OnRawKeyEvent;
+            Device.RegisterDevice(UsagePage.Generic, UsageId.GenericKeyboard, DeviceFlags.Remove);
```

Comment:
Clear _platform and Game references.
Clear the windows list in Mouse.
Remove KeyboardInput event handler and try to unregister raw keyboard input.  This is not documented, but some forum posts suggest it is the right thing to do.